packages/gatsby-plugin-cms/src/node-api/types.ts

```diff
@@ -0,0 +1,33 @@
+export interface Block {
+  name: string
+  props: Record<string, unknown>
+}
+
+export interface RemotePageContent {
+  remoteId: string
+  name: string
+  type: string
+  builderId: 'faststore'
+  author: {
+    id: string
+    email: string
+  }
+  lastUpdatedAt: string
+  blocks: Block[]
+  beforeBlocks: Block[]
+  afterBlocks: Block[]
+  extraBlocks: Array<{
+    name: string
+    blocks: Block[]
+  }>
+}
+
+export type PageContent = {
+  id: string
+  name: string
+  sections: Block[]
+} & {
+  [name: string]: {
+    [name: string]: Block['props']
+  }
+}
```

Comment:
```suggestion
} & Record<string, Record<string, Block['props']>>
```
But i prefer to name it as a tree/node because the nature of this content seems to be a tree.

Reply:
Why so? Isn't the first one more descriptive? It says the name prop of the cms will became the key and so on. Also, this is not a tree, it's just a two level construct

Reply:
So, IMHO the first one has the same meaning as the second, but the second is less verbose.